test/blocks/index.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import * as blocks from '@wordpress/blocks';
+
+describe( 'blocks API', () => {
+	// TODO: We probably want a way to undo this, and split this logic out into
+	// separate tests (clearing require.cache for example, but this probably
+	// won't work with webpack)
+	before( () => {
+		blocks.registerBlock( 'core/test-block' );
+		const blockSettings = { settingName: 'settingValue' };
+		blocks.registerBlock( 'core/test-block-with-settings', blockSettings );
+		// This is tested later: `registerBlock` should store a copy of its input
+		blockSettings.mutated = true;
+	} );
+
+	describe( 'validateBlockSlug', () => {
+		it( 'should reject numbers', () => {
+			expect(
+				() => blocks.validateBlockSlug( 999 )
+			).to.throw( /^Block slugs must contain a namespace prefix/ );
+		} );
+
+		it( 'should reject blocks without a namespace', () => {
+			expect(
+				() => blocks.validateBlockSlug( 'doing-it-wrong' )
+			).to.throw( /^Block slugs must contain a namespace prefix/ );
+		} );
+
+		it( 'should reject blocks with invalid characters', () => {
+			expect(
+				() => blocks.validateBlockSlug( 'still/_doing_it_wrong' )
+			).to.throw( /^Block slugs must contain a namespace prefix/ );
+		} );
+
+		it( 'should accept valid block names', () => {
+			expect(
+				() => blocks.validateBlockSlug( 'my-plugin/fancy-block-4' )
+			).not.to.throw();
+		} );
+	} );
+
+	// TODO: registerBlock tests
```

Comment:
I'm not really a fan of the approach though, it should be tested explicitly.  I think @aduth's suggestion `unregisterBlock` is a decent improvement.